File: ruby/6kyu/wich_are_in.rb
=begin
link: https://www.codewars.com/kata/550554fd08b86f84fe000a58
=end

def in_array(array1, array2)
  array1.select{ |x1| array2.any?{ |x2| x2.include?(x1) } }.sort
end

File: golang/6kyu/wich_are_in.go
package kata

import (
	"sort"
	"strings"
)

// https://www.codewars.com/kata/550554fd08b86f84fe000a58

func InArray(array1 []string, array2 []string) []string {
	hash := make(map[string]int)
	value := make([]string, 0)

	for _, w2 := range array2 {
		for _, w1 := range array1 {
			if strings.Contains(w2, w1) {

				if hash[w1] += 1; hash[w1] == 1 {
					value = append(value, w1)
				}

			}
		}
	}
	sort.Strings(value)

	return value
}
